yq: 2.8.1 -> 2.9.2

JohnAZoidberg opened the pull request:
###### Motivation for this change
Lots of new releases in the meantime: https://github.com/kislyuk/yq/releases

###### Things done

Tested with `./result/bin/yq . somefile.yml` and `./result/bin/xq . somefile.xml`.
There's not really much to break by altering the first argument since it just converts the file to JSON and pipes that including the first argument to jq.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @womfoo

JohnAZoidberg committed:
yq: 2.8.1 -> 2.10.1

Co-authored-by: Drew Risinger <drewrisinger@users.noreply.github.com>

JohnAZoidberg commented:
The tests fail now with a different error:

```
Run tests sequentially                                     
0:00:00 load avg: 1.74 [1/1] test
test test crashed -- Traceback (most recent call last):
  File "/nix/store/gpnm7i19lpj8p43mjrdw03d0hjalmskl-python3-3.7.5/lib/python3.7/test/libregrtest/runtest.py", line 270, in _runtest_inner
    refleak = _runtest_inner2(ns, test_name)
  File "/nix/store/gpnm7i19lpj8p43mjrdw03d0hjalmskl-python3-3.7.5/lib/python3.7/test/libregrtest/runtest.py", line 221, in _runtest_inner2
    the_module = importlib.import_module(abstest)
  File "/nix/store/gpnm7i19lpj8p43mjrdw03d0hjalmskl-python3-3.7.5/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'test.test'

test failed                                                

== Tests result: FAILURE ==                                

1 test failed:                                             
    test                                                   

Total duration: 57 ms                                      
Tests result: FAILURE
```

Looks like we could fix it by calling the testing framework properly but I don't have the time to figure that out, right now.

JohnAZoidberg commented:
ping @womfoo

drewrisinger reviewed (changes requested):
Run tests. Tested working with 

```nix
{ lib, buildPythonApplication, fetchPypi, pyyaml, xmltodict, jq, argcomplete, flake8, coverage, pytest, toml }:

buildPythonApplication rec {
  pname = "yq";
  version = "2.10.0";

  propagatedBuildInputs = [ pyyaml xmltodict jq argcomplete ];

  doCheck = true;

  checkInputs = [
    pytest
    coverage
    flake8
    jq
    toml
  ];

  checkPhase = ''
    pytest ./test/test.py
  '';
...
```
and
```nix
  yq = callPackage ../development/tools/yq {
    inherit (python3Packages) buildPythonApplication fetchPypi pyyaml xmltodict argcomplete pytest flake8 coverage toml;
  };
```

I recommend breaking arguments into one line per arg for appearance.

JohnAZoidberg commented:
Thanks @drewrisinger, I added you as co-author and updated to the latest version from yesterday.

Tested again with ./result/bin/yq . somefile.yml and ./result/bin/xq . somefile.xml.

cc @womfoo

drewrisinger commented on pkgs/top-level/all-packages.nix:
```diff
@@ -11071,7 +11071,9 @@ in
   yq = callPackage ../development/tools/yq {
-    inherit (python3Packages) buildPythonApplication fetchPypi pyyaml xmltodict;
+    inherit (python3Packages)
+    buildPythonApplication fetchPypi pyyaml xmltodict argcomplete pytest
+    coverage flake8 toml;
```
```suggestion
    buildPythonApplication fetchPypi argcomplete pyyaml xmltodict pytest coverage flake8 toml;
```

drewrisinger commented on pkgs/development/tools/yq/default.nix:
```diff
@@ -1,17 +1,31 @@
+   toml
+  ];
+
+  checkPhase = ''
+    pytest ./test/test.py
+  '';
```
```suggestion
  checkPhase = "pytest ./test/test.py";
```

drewrisinger commented on pkgs/development/tools/yq/default.nix:
```diff
@@ -1,17 +1,31 @@
-{ lib, buildPythonApplication, fetchPypi, pyyaml, xmltodict, jq }:
+{ lib, buildPythonApplication, fetchPypi
+, pyyaml, xmltodict, jq, argcomplete
+, pytest, coverage, flake8, toml
+}:
```
Break into separate lines & sort by name, roughly:
```suggestion
{ lib
, buildPythonApplication
, fetchPypi
, argcomplete
, pyyaml
, xmltodict
# Test inputs
, coverage
, flake8
, jq
, pytest
, toml
}:
```

drewrisinger reviewed (changes requested):
Small formatting suggestions

Mic92 commented:
Result of `nixpkgs-review pr 73424` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- yq
</details>

bhipple commented:
Would be nice to get this cleaned up prior to the 20.09 branch off, and updated to the latest at https://github.com/kislyuk/yq/blob/develop/Changes.rst

drewrisinger commented:
@bhipple when is that branch-off roughly?

bhipple commented:
Hasn't been announced yet, but you can catch up with @jonringer and follow https://discourse.nixos.org/t/upcoming-20-09-release-branch-off/8429/4 for the latest news.

Mic92 committed:
Apply suggestions from code review

Co-authored-by: Drew <drewrisinger@users.noreply.github.com>

Mic92 commented:
Result of `nixpkgs-review pr 73424` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- yq
</details>
